index/scorch/scorch.go

```diff
@@ -77,6 +77,12 @@ type Scorch struct {
 	segPlugin SegmentPlugin
 
 	spatialPlugin index.SpatialAnalyzerPlugin
+
+	// keeps track of segments scheduled for backup. Each segment's filename maps to
+	// the count of backup schedules. Segments with non-zero counts are protected
+	// from removal by the cleanup operation. Counts decrement upon successful backup,
+	// allowing removal of segments with zero or absent counts.
+	backupScheduled map[string]uint
```

Comment:
Naming is not consistent - if you're using the word "copy" for the interface, let's use the same here.

Reply:
Would you add a comment above that this needs to be accessed within lock - perhaps best move it to line 58 too , indicating access needs rootLock.